bench-tps/src/bench.rs

```diff
@@ -134,23 +135,34 @@ pub fn do_bench_tps<T>(
 
     // generate and send transactions for the specified duration
     let start = Instant::now();
+    let mut blockhash_time = Instant::now();
     let mut reclaim_lamports_back_to_source_account = false;
     let mut i = keypair0_balance;
+    let mut blockhash = Hash::default();
     while start.elapsed() < duration {
-        let balance = client.get_balance(&id.pubkey()).unwrap_or(0);
-        metrics_submit_lamport_balance(balance);
-
         // ping-pong between source and destination accounts for each loop iteration
         // this seems to be faster than trying to determine the balance of individual
         // accounts
         let len = tx_count as usize;
+        let new_blockhash = client.get_recent_blockhash().unwrap_or_default();
```

Comment:
Would `get_new_blockhash` here be any cleaner?

Reply:
And seems like if we can't get a blockhash, just ignoring it with unwrap_or_default seems like the wrong thing to do.. either we cannot connect to the network to get a blockhash at all and we should just panic, or we do something like try to look at other nodes.

Reply:
Yes definitely! Thanks!

Reply:
The network has stalled but aside from notifying the user I don't know what else it can do. A panic here results in many many panics on testnet (keeps restarting the client).

Reply:
I guess `get_new_blockhash` will panic so it'll be fatal when I switch to that

Reply:
Well, need to figure out the source of not being able to fetch it. If it's just not waiting long enough, then maybe the timeout should be increased. Maybe we should try switching to another node.

Reply:
I see what you mean. Yeah it's waiting plenty long but the network just stalls completely some times. Switching to another node might work but in our current testnets this should _never_ happen.